compile loadsave.h as C

**Describe the bug**
We venture further upon the quest to compile the DevilutionX headers as C. With #446, #447, #448 and #450, we can now start to compile most headers as C. However, there is one header that still relies on C++ specific treatment, and that is `loadsave.h`.

When we try to compile `loadsave.h` as C we arrive at the following error:

```
./Source/loadsave.h:36:6: error: conflicting types for 'SaveItem'
void SaveItem(int i);
```

The reason is simple, as is the solution. The reason is simply that in C++ the type is part of unique qualification of identifiers (which is why identifiers are mangled in C++), whereas in C, only the function name is part of the identifier. As such, we get a identifier collision for `SaveItem` which has multiple definitions with different types.

From `loadsave.h`:
```cpp
void SaveItem(ItemStruct *pItem);
void SaveItem(int i);
```

The solution would simply be to rename these definitions so they no longer share the same name.

```cpp
void SaveItem(ItemStruct *pItem);
void SaveItem(int i);
```

**Edit:** upon a closer look, it seems like `void SaveItem(int i);` is unused. I'll prepare a PR to remove it from the header file.

**Reproduce:**

The easiest way to reproduce this issue is simply to uncomment `extern "C"` in `diablo.h` and try to compile.

--- a/Source/loadsave.h
+++ b/Source/loadsave.h
@@ -33,7 +33,6 @@ void SavePlayer(int i);
 void SaveMonster(int i);
 void SaveMissile(int i);
 void SaveObject(int i);
-void SaveItem(int i);
 void SavePremium(int i);
 void SaveQuest(int i);
 void SaveLighting(int i);

